sklbench/utils/special_params.py

```diff
@@ -0,0 +1,246 @@
+        for i, dataset in enumerate(datasets):
+            if is_special_value(dataset):
+                dataset = dataset.replace(SP_VALUE_STR, '')
+                if dataset == 'all_named':
+                    datasets[i] = all_named_datasets
+        datasets = flatten_list(datasets, ensure_type_homogeneity=True)
+        set_bench_case_value(template, 'data:dataset', datasets)
+
+    return template
+
+
+def assign_case_special_values_on_generation(bench_case: BenchCase) -> BenchCase:
+    # sklearn.datasets.make_classification: n_informative as ratio of n_features
+    n_informative = get_bench_case_value(
+        bench_case, 'data:generation_kwargs:n_informative')
+    if is_special_value(n_informative):
+        n_informative = float(n_informative.replace(SP_VALUE_STR, ''))
+        if n_informative <= 0.0 or n_informative >= 1.0:
+            raise ValueError(f'Wrong special value "{n_informative}" for n_informative')
+        n_features = get_bench_case_value(bench_case, 'data:generation_kwargs:n_features')
+        if n_features is None:
+            raise ValueError(
+                '"n_features" is not specified for special value of "n_informative"')
+        set_bench_case_value(bench_case, 'data:generation_kwargs:n_informative',
+                             ceil(n_informative * n_features))
+    # taskset
+    taskset = get_bench_case_value(bench_case, 'bench:taskset')
+    if is_special_value(taskset):
+        taskset = taskset.replace(SP_VALUE_STR, '')
+        # special value format for numa nodes: "numa:{numa_node_0}[|{numa_node_1}...]"
+        if taskset.startswith('numa'):
+            numa_nodes = list(map(int, taskset.split(':')[1].split('|')))
+            numa_cpus_conf = get_numa_cpus_conf()
+            cpus = []
+            for numa_node in numa_nodes:
+                cpus += numa_cpus_conf[numa_node]
+            if len(cpus) == 0:
+                raise ValueError(
+                    f'Specification of {numa_nodes} NUMA nodes resulted in zero cpus.')
+            taskset = str(cpus)[1:-1].replace(' ', '')
+            set_bench_case_value(bench_case, 'bench:taskset', taskset)
+    return bench_case
+
+
+def get_ratio_from_n_jobs(n_jobs: str) -> float:
+    args = n_jobs.split(':')
+    if len(args) == 1:
+        return 1.0
+    elif len(args) == 2:
+        return float(args[1])
+    else:
+        raise ValueError(f'Wrong arguments {args} in "n_jobs" special value')
+
+
+def assign_case_special_values_on_run(
+        bench_case: BenchCase,
+        x_train,
+        y_train,
+        x_test,
+        y_test,
+        data_description: Dict):
+    library = get_bench_case_value(bench_case, 'algorithm:library', None)
+    estimator = get_bench_case_value(bench_case, 'algorithm:estimator', None)
+    # device-related parameters assignment
+    device = get_bench_case_value(bench_case, 'algorithm:device', 'default')
+    if device != 'default':
```

Comment:
There should be a warning that by specifying 'cpu' or 'gpu' will use 'device_offload' methods within sklearnex by default, which isn't how sklearnex operates by default.  This impacts the operation of the configs/sklearn/trees.json.  Maybe:

if device != 'default' and device != 'cpu':

Reply:
Fixed: now `cpu` setting doesn't affect target offload for sklearnex.